apis/v1alpha1/types.go

```diff
@@ -0,0 +1,661 @@
+// or Working with Amazon Elastic Inference on Amazon ECS (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-inference.html)
+// in the Amazon Elastic Container Service Developer Guide
+type ECSResourceRequirement struct {
+	Type  *string `json:"type_,omitempty"`
+	Value *string `json:"value,omitempty"`
+}
+
+// The overrides that are associated with a task.
+type ECSTaskOverride struct {
+	ContainerOverrides []*ECSContainerOverride `json:"containerOverrides,omitempty"`
+	CPU                *string                 `json:"cpu,omitempty"`
+	// The amount of ephemeral storage to allocate for the task. This parameter
+	// is used to expand the total amount of ephemeral storage available, beyond
+	// the default amount, for tasks hosted on Fargate. For more information, see
+	// Fargate task storage (https://docs.aws.amazon.com/AmazonECS/latest/userguide/using_data_volumes.html)
+	// in the Amazon ECS User Guide for Fargate.
+	//
+	// This parameter is only supported for tasks hosted on Fargate using Linux
+	// platform version 1.4.0 or later. This parameter is not supported for Windows
+	// containers on Fargate.
+	EphemeralStorage              *ECSEphemeralStorage               `json:"ephemeralStorage,omitempty"`
+	ExecutionRoleARN              *string                            `json:"executionRoleARN,omitempty"`
+	InferenceAcceleratorOverrides []*ECSInferenceAcceleratorOverride `json:"inferenceAcceleratorOverrides,omitempty"`
+	Memory                        *string                            `json:"memory,omitempty"`
+	TaskRoleARN                   *string                            `json:"taskRoleARN,omitempty"`
+}
+
+// Filter events using an event pattern. For more information, see Events and
+// Event Patterns (https://docs.aws.amazon.com/eventbridge/latest/userguide/eventbridge-and-event-patterns.html)
+// in the Amazon EventBridge User Guide.
+type Filter struct {
+	Pattern *string `json:"pattern,omitempty"`
+}
+
+// The collection of event patterns used to filter events. For more information,
+// see Events and Event Patterns (https://docs.aws.amazon.com/eventbridge/latest/userguide/eventbridge-and-event-patterns.html)
+// in the Amazon EventBridge User Guide.
+type FilterCriteria struct {
+	Filters []*Filter `json:"filters,omitempty"`
+}
+
+// The Secrets Manager secret that stores your broker credentials.
+type MQBrokerAccessCredentials struct {
+	// // Optional SecretManager ARN which stores the database credentials
+	BasicAuth *string `json:"basicAuth,omitempty"`
+}
+
+// The Secrets Manager secret that stores your stream credentials.
+type MSKAccessCredentials struct {
+	// // Optional SecretManager ARN which stores the database credentials
+	ClientCertificateTLSAuth *string `json:"clientCertificateTLSAuth,omitempty"`
+	// // Optional SecretManager ARN which stores the database credentials
+	SaslScram512Auth *string `json:"saslScram512Auth,omitempty"`
```

Comment:
I think we need an new initialism added to our list, here? `SASL` and `SCRAM`

Reply:
I see, let me see if I can get this right in a PR...

Reply:
https://github.com/aws-controllers-k8s/pkg/pull/16